src/common/lru.c

```diff
@@ -258,16 +258,16 @@ lru_ref_release_internal(struct daos_lru_cache *lcache, struct daos_llink *llink
 	D_ASSERT(lcache != NULL && llink != NULL && llink->ll_ref > 1);
 	D_ASSERT(d_list_empty(&llink->ll_qlink));
 
-	lru_hop_rec_decref(&lcache->dlc_htable, &llink->ll_link);
-
-	if (wait && llink->ll_ref > 1) {
+	if (wait && llink->ll_ref > 2) {
 		D_ASSERT(llink->ll_evicting == 0);
 		llink->ll_evicting = 1;
 		lcache->dlc_ops->lop_wait(llink);
 		llink->ll_evicting = 0;
 		llink->ll_evicted = 1;
 	}
 
+	lru_hop_rec_decref(&lcache->dlc_htable, &llink->ll_link);
```

Comment:
It looks to me a simpler (cleaner) fix is to replace the daos_lru_ref_release() in daos_lru_ref_hold() with lru_hopd_rec_decref(), what do you think?

Reply:
As discussed, we should use lru_hopd_rec_decref() here, otherwise wakeup() might be called twice here.